### PR TITLE
User story 45: Merchant can create new items

### LINKED ISSUE #76: User Story 45, Merchant adds an item
As a merchant
When I visit my items page
I see a link to add a new item
When I click on the link to add a new item
I see a form where I can add new information about an item, including:
- the name of the item, which cannot be blank
- a description for the item, which cannot be blank
- a thumbnail image URL string, which CAN be left blank
- a price which must be greater than $0.00
- my current inventory count of this item which is 0 or greater

When I submit valid information and submit the form
I am taken back to my items page
I see a flash message indicating my new item is saved
I see the new item on the page, and it is enabled and available for sale
If I left the image field blank, I see a placeholder image for the thumbnail

### PR DESCRIPTION
**Pull request by DanielEFrampton**
## Description
Resolves #76 
Fixed more Capybara warnings
Some refactoring of Merchant items_controller index action

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which maintains existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## RSpec Results
```
Finished in 11.11 seconds (files took 2.38 seconds to load)
233 examples, 0 failures

Coverage report generated for RSpec to /Users/dframpton/turing/2module/group_projects/monster_shop_part_1/coverage. 2740 / 2743 LOC (99.89%) covered.
```

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -52,9 +52,9 @@
 
   namespace :merchant do
     get '/', to: 'dashboard#index'
-    get '/items', to: 'items#index'
     patch '/items/:id/activation', to: 'items#update'
 
+    resources :items, only: [:index, :new, :create]
```
Comment (mel-rob) [inline config/routes.rb:57]: yay resources!!

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -34,6 +34,7 @@
 
   describe 'relationships' do
     it {should have_many :orders}
+    it {should belong_to(:merchant).optional}
```
Comment (mel-rob) [inline spec/models/user_spec.rb:37]: What does this `.optional` do?

**Review by mel-rob — APPROVED** [review]
Great work!

File: app/controllers/merchant/items_controller.rb [hunk 3]
```diff
@@ -1,17 +1,38 @@
 class Merchant::ItemsController < Merchant::BaseController
 
   def index
-    @merchant = Merchant.first
```
Comment (philjdelong) [inline app/controllers/merchant/items_controller.rb:4]: thank you sir. meant to ask about this earlier...got quite a few interesting error messages

File: app/views/merchant/items/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<h1>Create New Item</h1>
```
Comment (philjdelong) [inline app/views/merchant/items/new.html.erb:1]: "Hit 'em with a little .form_for"

File: spec/features/merchant/items/index_spec.rb [hunk 5]
```diff
@@ -21,7 +21,7 @@
         expect(page).to have_content("Price: $#{@tire.price}")
         expect(page).to have_css("img[src*='#{@tire.image}']")
         expect(page).to have_content("Active")
-        expect(page).to_not have_content(@tire.description)
```
Comment (philjdelong) [inline spec/features/merchant/items/index_spec.rb:24]: I went back and looked at this earlier after we had talked briefly and I did the same thing. Removed the condition from the view as well

**Review by philjdelong — APPROVED** [review]

**Pull request merged**